### PR TITLE
Added file closing functionality and added another unicode validation

### PR DESCRIPTION
**Pull request by malun22**
In case an error is raised when creating a GBX object the file connection is not being closed which is required, if you want to continue working on the file (e.g. deleting it).

Also when I tried to import an exe file, it failed raising the GbxLoadError, because it could not decode as utf-8. That's why I added another try block to potentially raise an GbxLoadError.

### CODE REVIEW

File: pygbx/gbx.py [hunk 1]
```diff
@@ -135,6 +149,11 @@ def __init__(self, obj):
         bp = ByteReader(self.data)
         self._read_node(self.class_id, -1, bp)
 
+    """Closes the current file connection"""
+
+    def close(self):
```
Comment (donadigo) [inline pygbx/gbx.py:154]: Also note that `self.f` may not even exist when calling this method when constructing a Gbx instance with raw byte data. This should be handled in some way, either doing nothing or throwing an exception.

**Review by donadigo — CHANGES_REQUESTED** [review]
Hello! Thank you for your work. Are code style changes intentional here? I see you have replaced single with double quotes and changed indentation in some places which I'd like to avoid for this PR.

As for the actual change: that's looking good, but if we have a `close` method at this point, it would be logical to have the Gbx class implement a context manager so that this is also possible:
```
with Gbx(path) as gfile:
    # do something with gfile

# gfile is automatically closed here
```

I also commented on the diff.

**Comment by malun22** [comment]
The style changes are not intended, I am sorry. My auto formatter created them. I though about deleting the pull request again and create a new one without those. Should I?

I did not see that f might not exists at this point. If it's not existent, doing nothing should be fine.

I have not worked with contexts yet, but sounds interesting.

**Comment by donadigo** [comment]
Sure thing. Do what's most comfortable for you,

**Pull request closed without merging**